rules/always-return.js

```diff
@@ -244,6 +282,23 @@ module.exports = {
           return
         }
 
+        if (ignoreAssignmentVariable.length && isLastCallback(node)) {
+          let hasIgnoredAssignment = false
+
+          //istanbul ignore else
+          if (node.body?.type === 'BlockStatement') {
+            node.body.body.forEach((statement) => {
+              if (isIgnoredAssignment(statement, ignoreAssignmentVariable)) {
+                hasIgnoredAssignment = true
+              }
+            })
+          }
+
+          if (hasIgnoredAssignment) {
+            return
+          }
```

Comment:
**suggestion:** We can cut out some iterations here, by just returning in the middle of the loop
```suggestion
          //istanbul ignore else
          if (node.body?.type === 'BlockStatement') {
            for(const statement of node.body.body) {
              if (isIgnoredAssignment(statement, ignoreAssignmentVariable)) {
                return
              }
            }
          }
```